catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceSource.java

```diff
@@ -0,0 +1,386 @@
+                query.getStartIndex() - 1,
+                query.getPageSize(),
+                includeArchivedSpaces);
+
+        InputStream stream = null;
+        Object entityObj = confluenceResponse.getEntity();
+        if (entityObj != null) {
+            stream = (InputStream) entityObj;
+        }
+        if (Response.Status.OK.getStatusCode() != confluenceResponse.getStatus()) {
+            String error = "";
+            try {
+                if (stream != null) {
+                    error = IOUtils.toString(stream);
+                }
+            } catch (IOException ioe) {
+                LOGGER.debug("Could not convert error message to a string for output.", ioe);
+            }
+            throw new UnsupportedQueryException(String.format(
+                    "Received error code from remote source (status %s ): %s",
+                    confluenceResponse.getStatus(),
+                    error));
+        }
+
+        try {
+
+            List<Result> results = transformer.transformConfluenceResponse(stream)
+                    .stream()
+                    .map(this::getUpdatedResult)
+                    .collect(Collectors.toList());
+
+            return new SourceResponseImpl(request, results);
+        } catch (IOException | CatalogTransformerException e) {
+            throw new UnsupportedQueryException("Exception processing results from confluence");
+        }
+    }
+
+    @Override
+    public Set<ContentType> getContentTypes() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public ResourceResponse retrieveResource(URI uri, Map<String, Serializable> arguments)
+            throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(password)) {
+            arguments.put(USERNAME, username);
+            arguments.put(PASSWORD, password);
+        }
+        return resourceReader.retrieveResource(uri, arguments);
+    }
+
+    @Override
+    public Set<String> getSupportedSchemes() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Set<String> getOptions(Metacard metacard) {
+        return Collections.emptySet();
+    }
+
+    public void setAvailabilityPollInterval(long availabilityPollInterval) {
+        this.availabilityPollInterval = availabilityPollInterval;
+    }
+
+    public void setEndpointUrl(String endpointUrl) {
+        if (endpointUrl != null) {
+            endpointUrl = endpointUrl.trim();
+        }
+        this.endpointUrl = PropertyResolver.resolveProperties(endpointUrl);
+        init();
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setPassword(String password) {
+        String updatedPassword = password;
+        if (encryptionService != null) {
+            updatedPassword = encryptionService.decryptValue(password);
+        }
+        this.password = updatedPassword;
```

Comment:
I know the scope here is tiny, there aren't any variable accessors and I assume the garbage collector comes in quick, but is there any security issue with this decrypted password string being package private?